bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/connector/Executor.java

```diff
@@ -157,6 +157,49 @@ public void notifyFailure(ErrorValue error) {
         }
     }
 
+    /**
+     * This method will invoke Ballerina function in a non-blocking manner.
+     *
+     * @param scheduler   The Current scheduler
+     * @param classLoader The classLoader
+     * @param orgName     The org which the package belongs to
+     * @param packageName The package which the class belongs to
+     * @param className   The class name or the file name in which the function resides
+     * @param methodName  The method name which is to be the invokable unit
+     * @param callback    The callback to be executed when the execution completes.
+     * @param paramValues The Parameters to be passed to the invokable unit
+     */
+    public static void executeFunctionAsync(Scheduler scheduler, ClassLoader classLoader, final String orgName,
```

Comment:
Since this new executeFunctionAsync takes callback as a param, existing 'executeFunctionAsync'  can call this function passing the callback with CountDownLatch. So we do not need to duplicate common logic

Reply:
Removed this code segment as we decided not to make this async.